libs/deepagents-cli/deepagents_cli/git_context.py

```diff
@@ -0,0 +1,178 @@
+"""Middleware for injecting git state and directory context into system prompt."""
```

Comment:
I was thinking about making the middleware a bit more generic so we end up with a single middleware that is responsible for injecting local context.

The context will be:

1. CWD
2. list of current files (truncated)
3. tree (truncated)
4. git information

----

Basically I'm thinking that we don't gain much by splitting these into different middlewares since we'll be using all features at the same time or not at all.

---

The only complication is what do we do with sandboxes. The sandboxes do have an `exec` method, so could be nice to standardize around that? (but we could add support for this in a follow up PR)